tools/gulp/tasks/docs.ts

```diff
@@ -177,3 +180,197 @@ function createTagNameAliaser(classPrefix: string) {
     return this;
   };
 }
+
+interface ExampleMetadata {
+  component: string;
+  filename: string;
+  id: string;
+  title: string;
+  additionalComponents?: any[];
+  additionalFiles?: string;
+  selectorName?: string;
+}
+
+/**
+ * Builds the template for the examples module
+ * @param results
+ */
+const buildTemplate = (results: ExampleMetadata[]): string => {
+  // create the import statements
+  const imports = (res: ExampleMetadata): string => {
+    const components = [res.component];
+
+    if (res.additionalComponents) {
+      components.push(...res.additionalComponents);
+    }
+
+    return `import {${components.join(',')}} from '${res.filename}';
+  `;
+  };
+
+  // create the examples metadat
+  const examples = (res: ExampleMetadata): string => {
+    return `'${res.id}': {
+        title: '${res.title}',
+        component: ${res.component},
+        additionalFiles: ${res.additionalFiles ? ('[' + res.additionalFiles + ']') : 'undefined'},
+        selectorName: ${res.selectorName ? ("'" + res.selectorName + "'") : 'undefined'}
+    },
+    `;
+  };
+
+  // create the examples component list
+  const list = (res: ExampleMetadata): string => {
+    const components = [res.component];
+
+    if (res.additionalComponents) {
+      components.push(...res.additionalComponents);
+    }
+    return `${components.join(',')},
+    `;
+  };
+
+  return `
+  import {NgModule} from '@angular/core';
```

Comment:
Actually let's see how small it gets when moving the material modules into their own file; maybe then it will be short enough to leave as a string.